components/dashboard/ActivityInputDialog.js

```diff
@@ -0,0 +1,356 @@
+  },
+  textField: {
+    marginTop: theme.spacing(2),
+    marginBottom: theme.spacing(1),
+  },
+  toggleButton: {
+    marginTop: theme.spacing(1),
+    marginBottom: theme.spacing(1),
+  },
+}));
+
+const ACTIVITY_TYPES = [
+  'Searched for job openings (e.g., Indeed, LinkedIn, Monster, CareerBuilder, etc.)',
+  'Completed job application',
+  'Created, revised, or customized marketing materials (e.g., resume, cover letter, business card, LinkedIn profile, LinkedIn post, etc.)',
+  'Prepared for interview',
+  'Attend networking event/job fair',
+  'Virtual networking interaction (via email, LinkedIn, etc.)',
+  'Meeting/Call with contact',
+  'Researched target company/industry',
+  'Researched contacts at target company',
+  'Other',
+];
+
+const TIME_SPENT_TYPE = [
+  {
+    label: '30 minutes',
+    value: 30,
+  },
+  {
+    label: '60 minutes',
+    value: 60,
+  },
+  {
+    label: '90 minutes',
+    value: 90,
+  },
+  {
+    label: '120+ minutes',
+    value: 120,
+  },
+];
+
+const DIFFICULTY_LEVEL = ['Easy', 'Medium', 'Hard'];
+
+const FEELINGS = [
+  'Confident',
+  'Discouraged',
+  'Frustrated',
+  'Optimistic',
+  'Motivated',
+  'Overwhelmed',
+];
+
+// random shuffle of Feeling types.
+FEELINGS.sort(() => {
+  return 0.5 - Math.random();
+});
+
+const activityFormValues = {
+  activityType: ACTIVITY_TYPES[0],
+  description: '',
+  dateCompleted: new Date(),
+  timeSpentInMinutes: TIME_SPENT_TYPE[0].value,
+  difficultyLevel: DIFFICULTY_LEVEL[0],
+  activityFeeling: [],
+  whyIfeelThisWay: null,
+};
+
+const isEmpty = s => {
+  return s === undefined || s === null || s === '';
+};
+
+function ActivityInputDialog({ show, onClose }) {
+  const classes = useActivityDialogStyles();
+  const formId = 'activity-input';
+  const { userDocRef } = useAuth();
+  const [error, setError] = useState();
+  const [submitting, setSubmitting] = useState(false);
+  const [success, setSuccess] = useState(false);
+  const [formValues, setFormValues] = useState(activityFormValues);
+  const [formErrors, setFormErrors] = useState({});
+
+  const isFormValid = () => {
+    setFormErrors({});
+    if (isEmpty(formValues.description)) {
+      setFormErrors({ ...formErrors, description: 'Field is required.' });
+      return false;
+    }
+    return true;
+  };
+
+  const handleSave = () => {
+    const data = {
+      timestamp: new Date(),
+      ...formValues,
+    };
+    setError();
+
+    if (isFormValid()) {
+      setSubmitting(true);
+      userDocRef
+        .collection('activityLogEntries')
+        .add(data)
+        .then(() => {
+          setSubmitting(false);
+          setSuccess(true);
+        })
+        .catch(err => {
+          setSubmitting(false);
+          setError(err.message);
+        });
+    }
+  };
+
+  const resetComponent = () => {
+    setError();
+    setSuccess(false);
+    setSubmitting(false);
+    setFormValues(activityFormValues);
+    setFormErrors({});
+  };
+
+  return (
+    <Dialog
```

Comment:
Thoughts on using `withMobileDialog` like the existing dialogs to go full-screen in mobile?

Reply:
Added.